server/src/test/java/org/elasticsearch/persistent/PersistentTasksCustomMetaDataTests.java

```diff
@@ -228,7 +242,66 @@ public void testBuilder() {
             assertEquals(changed, builder.isChanged());
             persistentTasks = builder.build();
         }
+    }
+
+    public void testMinVersionSerialization() throws IOException {
+        PersistentTasksCustomMetaData.Builder tasks = PersistentTasksCustomMetaData.builder();
+
+        Version minVersion = getFirstVersion();
+        final Version streamVersion = randomVersionBetween(random(), minVersion, getPreviousVersion(Version.CURRENT));
+        tasks.addTask("test_compatible_version", TestPersistentTasksExecutor.NAME,
+            new TestParams(null, randomVersionBetween(random(), minVersion, streamVersion),
+                randomBoolean() ? Optional.empty() : Optional.of("test")),
+            randomAssignment());
+        tasks.addTask("test_incompatible_version", TestPersistentTasksExecutor.NAME,
+            new TestParams(null, randomVersionBetween(random(), getNextVersion(streamVersion), Version.CURRENT),
+                randomBoolean() ? Optional.empty() : Optional.of("test")),
+            randomAssignment());
+        final BytesStreamOutput out = new BytesStreamOutput();
+        out.setVersion(streamVersion);
+        Set<String> features = new HashSet<>();
+        if (randomBoolean()) {
+            features.add("test");
+        }
+        if (randomBoolean()) {
+            features.add(TransportClient.TRANSPORT_CLIENT_FEATURE);
+        }
+        tasks.build().writeTo(out);
+
+        final StreamInput input = out.bytes().streamInput();
+        input.setVersion(streamVersion);
+        PersistentTasksCustomMetaData read =
+            new PersistentTasksCustomMetaData(new NamedWriteableAwareStreamInput(input, getNamedWriteableRegistry()));
+
+        Set<String> expectedIds = new HashSet<>(tasks.getCurrentTaskIds());
+        expectedIds.remove("test_incompatible_version");
```

Comment:
This is indirect. Why not `Collections.singleton("test_compatible_version")`? Then the local is not even needed and it saves a few lines too.

Reply:
Left over where I had more random tasks in there but I removed it due to other issues. Will simplify.